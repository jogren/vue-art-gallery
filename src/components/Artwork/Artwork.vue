<template>
  <div @click="toggleImage" @keyup="toggleImageOnKeyUp" class="Artwork-container" tabindex="0">
    <img v-if="image.showPicture" :src="image.image"/>
    <div v-else class="image-info-container">
      <p>Title: {{image.title}}</p>
      <p>Artist: {{image.artist}}</p>
      <p>Culture: {{image.culture}}</p>
      <p>Century: {{image.century}}</p>
    </div>
  </div>
</template>

<script>
export default {
  name: "Artwork",
  props: ["image"],
  methods: {
    toggleImage() {
      this.image.showPicture = !this.image.showPicture;
    },
    toggleImageOnKeyUp(e) {
      if(e.keyCode === 13) {
        this.image.showPicture = !this.image.showPicture;
      }
    }
  }
}
</script>

<style scoped>
.Artwork-container {
  height: 500px;
  width: 300px;
  margin: 10px;
}

.image-info-container {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.image-info-container p {
  margin-bottom: 5px;
}

img {
  height: 100%;
  width: 100%;
}
  
img:hover {
  animation: artwork_hover 1s;
  animation-timing-function: linear;
}
  
@keyframes artwork_hover {
  0% { transform: rotate(0deg); }
  25% {transform: rotate(2deg); }
  50% {transform: rotate(0deg); }
  75% {transform: rotate(-2deg); }
  100% {transform: rotate(0deg); }
  }

</style>