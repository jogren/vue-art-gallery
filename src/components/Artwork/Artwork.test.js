import { mount } from '@vue/test-utils'
import Artwork from './Artwork'

describe('Artwork', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Artwork, {
      propsData: {
        image: {
          showPicture: true
        }
      }
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = mount(Artwork, {
      propsData: {
        image: {
          showPicture: true
        }
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})