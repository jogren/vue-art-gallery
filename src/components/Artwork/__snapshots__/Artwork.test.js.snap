// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artwork renders correctly 1`] = `
<div
  class="Artwork-container"
>
  <img />
</div>
`;
